cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -67,26 +64,32 @@ public ApplicationsOperations(DefaultCloudFoundryOperations cloudFoundryOperatio
                         .builder()
                         .name(applicationSummary.getName())
                         .build());
+
     }
 
     /**
      * Deletes a specific application associated with the name <code>applicationName</code>.
      *
      * @param applicationName applicationName Name of an application.
+     * @throws NullPointerException when the applicationName is null
+     * @return Mono which can be subscribed on to trigger the removal of the app
      */
-    public void removeApplication(String applicationName) {
+    public Mono<Void> remove(String applicationName) {
+        checkNotNull(applicationName);
+
         DeleteApplicationRequest request = DeleteApplicationRequest
                 .builder()
                 .name(applicationName)
                 .build();
-        try {
-            this.cloudFoundryOperations.applications().delete(request).block();
-            Log.info("Application " + applicationName + " has been successfully removed.");
-        } catch (Exception e) {
-            Log.error(e.getMessage());
-        }
+
+        return this.cloudFoundryOperations.applications()
+                .delete(request)
+                .doOnSuccess(aVoid -> {
+                    Log.info("Application " + applicationName + " has been successfully removed.");
+                });
     }
 
+
     /**
      * Pushes the app to the cloud foundry instance specified within the cloud foundry operations instance
      *
```

Comment:
I think the documentation does not fit anymore. It should explain that the request for creating an application is returned. Also the behaviour when encountering non-existing services should be documented.
Again we should mention that the resulting Mono takes care of logging.

Reply:
Updated